ArduPlane/commands_logic.cpp

```diff
@@ -943,6 +963,70 @@ void Plane::exit_mission_callback()
     }
 }
 
+bool Plane::verify_landing_vtol_approach(const AP_Mission::Mission_Command &cmd)
+{
+    switch (vtol_approach_s.approach_stage) {
+        case LOITER_TO_ALT:
+            if (verify_loiter_to_alt()) {
+                Vector3f wind = ahrs.wind_estimate();
+                vtol_approach_s.approach_direction = (degrees(atan2f(-wind.y, -wind.x)));
+                gcs().send_text(MAV_SEVERITY_INFO, "Selected an approach path of %f", (double)vtol_approach_s.approach_direction);
+                // select target approach direction
+                // select detransition distance (add in extra distance if the approach does not fit in the required space)
+                // validate turn
+                vtol_approach_s.approach_stage = WAIT_FOR_BREAKOUT;
+            }
+            break;
+        case WAIT_FOR_BREAKOUT:
+            {
+                const float breakout_direction = radians(wrap_180(vtol_approach_s.approach_direction + 270));
+
+                nav_controller->update_loiter(cmd.content.location, aparm.loiter_radius, cmd.content.location.flags.loiter_ccw ? -1 : 1);
+
+                // breakout when within 5 degrees of the opposite direction
+                if (fabsf(ahrs.yaw - breakout_direction) < radians(5.0f)) {
+                    gcs().send_text(MAV_SEVERITY_INFO, "Starting VTOL land approach path");
+                    vtol_approach_s.approach_stage = APPROACH_LINE;;
+                    set_next_WP(cmd.content.location);
+                    // fallthrough
+                } else {
+                    break;
+                }
+                FALLTHROUGH;
+            }
+        case APPROACH_LINE:
+            {
+                // project an apporach path
+                Location start, end;
+                memcpy(&start, &cmd.content.location, sizeof(Location));
```

Comment:
why are these not structure assignments?

Reply:
Grabbed it from deepstall code which had used memcpy as well, will convert it, the compiler seems to do equally well with it either way.